### PR TITLE
Implement events functionality

### PR DESCRIPTION
**Pull request by sheshenk**
Basic implementation of events (no support for add, edit, delete, view, find yet).

- Created `Event` and `UniqueEventList` models with all their dependencies.
- Updated `AddressBook` models by adding the events list.
- Updated invalid messages for events.
- Added `JsonSerializableEvent` to support event storage.
- Updated all profile tests to correspond to updated `AddressBook`.
- Created an abstract `EventCommand` class.
- Added new event prefixes: `start/`, `end/`.
- Created an `EventCommandParser` with only default behaviour.

### CODE REVIEW

File: src/test/java/seedu/address/model/event/DateTimeTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Test
+    public void constructor_invalidDate_throwsIllegalArgumentException() {
+        String invalidDate = "";
+        assertThrows(IllegalArgumentException.class, () -> new DateTime(invalidDate));
+    }
+
+    @Test
+    public void isValidEmail() {
```
Comment (ShawnChew) [inline src/test/java/seedu/address/model/event/DateTimeTest.java:23]:
```suggestion
    public void isValidDateTime() {
```

Comments about the invalid datetime mentioned above could be applied here as well.

**Review by ShawnChew — COMMENTED** [review]
Really great job overall 👍 👍 

Since we don't have an add event currently, I couldn't really test if there were errors, but the test cases were well written, so there shouldn't be any problem. I compared it to the implementation of profiles, and there seems to be no issues.

However, I think we need to be able to add profiles to the events, and also we need to be able to check if the datetime overlaps (although this one is more relevant for adding new events, I think the datetime class should have some method to check it).

**Review by lesterong — COMMENTED** [review]
Great work! Just a couple of changes and the UI to hook up

File: src/main/java/seedu/address/model/event/DateTime.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    private static final Map<String, String> DATE_VALIDATION_REGEXS = new HashMap<>() {{
+            put("^\\d{1,2}-\\d{1,2}-\\d{4}$", "dd-MM-yyyy");
+            put("^\\d{1,2}-\\d{1,2}-\\d{2}$", "dd-MM-yy");
+            put("^\\d{4}-\\d{1,2}-\\d{1,2}$", "yyyy-MM-dd");
+            put("^\\d{1,2}/\\d{1,2}/\\d{4}$", "dd/MM/yyyy");
+            put("^\\d{1,2}/\\d{1,2}/\\d{2}$", "dd/MM/yy");
+            put("^\\d{4}/\\d{1,2}/\\d{1,2}$", "yyyy/MM/dd");
+            put("^\\d{1,2}\\s[a-z]{3}\\s\\d{4}$", "dd MMM yyyy");
+            put("^\\d{1,2}\\s[a-z]{4,}\\s\\d{4}$", "dd MMMM yyyy");
+        }};
```
[the remark this answers is not in the document]
Comment (sheshenk) [reply src/main/java/seedu/address/model/event/DateTime.java:32]: We'll resolve this in v1.4.

**Review by lesterong — APPROVED** [review]

**Pull request merged**